File: 24. Swap Nodes in Pairs
/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(next = head)
        prev, curr, foll = dummy, head, head
        while foll and foll.next:
            foll = foll.next
            prev.next = foll
            curr.next = foll.next
            foll.next = curr
            prev, curr, foll = curr, curr.next, curr.next
        return dummy.next

File: 24. Swap Nodes in Pairs
/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy = new ListNode(0, head), prev = dummy, curr = head, foll = head;
        while (foll != null && foll.next != null) {
            foll = foll.next;
            prev.next = foll;
            curr.next = foll.next;
            foll.next = curr;
            prev = curr;
            curr = curr.next;
            foll = curr;
        }
        return dummy.next;
    }
}